packages/eas-cli/src/utils/queries.ts

```diff
@@ -0,0 +1,138 @@
+import { confirmAsync, selectAsync } from '../prompts';
+import uniqBy from './expodash/uniqBy';
+
+const fetchMoreValue = '_fetchMore';
+
+export type PaginatedQueryResponse<QueryReturnType extends Record<string, any>> = {
+  queryResponse: QueryReturnType[];
+  queryResponseRawLength: number;
+};
+
+type BasePaginatedQueryArgs<QueryReturnType extends Record<string, any>> = {
+  pageSize: number;
+  offset: number;
```

Comment:
Nit: I didn't want to bring it up before but usually it's `limit` + `offset` or  `pageSize` + `pageNumber`. You're mixing both approaches.